src/odemis/util/transform.py

```diff
@@ -193,6 +193,98 @@ def _optimal_rotation(x, y):
     return numpy.dot(V.T, U.T)
 
 
+def to_physical_space(ji, shape, pixel_size=None):
+    """
+    Converts an image pixel index into a coordinate in physical space.
+
+    Images are displayed on a computer screen with the origin in the top-left
+    corner. The column-index `i` is used as the primary axis and the row-index
+    `j` is used as the secondary axis. This represents a left-handed coordinate
+    system. Physical coordinates are typically described in a right-handed
+    coordinate system.
+
+    Calculations should be done preferably in the right-handed physical
+    coordinate system. Calculations using mixed left-handed and right-handed
+    coordinates should be avoided.
+
+    `to_physical_space` converts an image pixel index into a coordinate in
+    physical space. The x-axis is aligned with the columns of the image, and
+    the y-axis with the rows. The direction of the y-axis is opposite to the
+    row-index, such that is increasing in the upward direction. The origin is
+    placed at the center of the image.
+
+
+              Image Pixel Indices                   Physical Coordinates
+
+      o--> i
+      | +---------+---------+---------+        +---------+---------+---------+
+    j V |         |         |         |        |         |         |         |
+        |  (0,0)  |  (0,1)  |  (0,2)  |        | (-1, 1) |  (0, 1) |  (1, 1) |
+        |         |         |         |        |         |         |         |
+        +---------+---------+---------+        +---------+---------+---------+
+        |         |         |         |        |         |         |         |
+        |  (1,0)  |  (1,1)  |  (1,2)  |        | (-1, 0) |  (0, 0) |  (1, 0) |
+        |         |         |         |        |         |         |         |
+        +---------+---------+---------+        +---------+---------+---------+
+        |         |         |         |        |         |         |         |
+        |  (2,0)  |  (2,1)  |  (2,2)  |        | (-1,-1) |  (0,-1) |  (1,-1) |
+        |         |         |         |    y ^ |         |         |         |
+        +---------+---------+---------+      | +---------+---------+---------+
+                                             o--> x
```

Comment:
Really nice! I am missing a note on the origin, which is the center of the first pixel in image space. That was a bottle neck for me to understand why the coordinates in spatial space can be mapped to non integer indices (e.g. (0,0) -> (-2. ,  3.5). A simple note could already hint the next reader to not get stuck here.